controllers/imagerepository_controller.go

```diff
@@ -60,13 +70,52 @@ type ImageRepositoryReconciler struct {
 	QuayOrganization string
 }
 
+func initMetrics() error {
+	buckets := getProvisionTimeMetricsBuckets()
+
+	// don't register it if it was already registered by another controller
+	if imageRepositoryProvisionTimeMetric != nil {
+		return nil
+	}
+
+	imageRepositoryProvisionTimeMetric = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: metricsNamespace,
+		Subsystem: metricsSubsystem,
+		Buckets:   buckets,
+		Name:      "image_repository_provision_time",
+		Help:      "The time in seconds spent from the moment of Image repository provision request to Image repository is ready to use.",
+	})
+
+	if err := metrics.Registry.Register(imageRepositoryProvisionTimeMetric); err != nil {
+		return fmt.Errorf("failed to register the image_repository_provision_time metric: %w", err)
+	}
+
+	return nil
+}
+
+func getProvisionTimeMetricsBuckets() []float64 {
+	return []float64{5, 10, 15, 20, 30, 60, 120, 300}
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *ImageRepositoryReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if err := initMetrics(); err != nil {
+		return err
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&imagerepositoryv1alpha1.ImageRepository{}).
 		Complete(r)
 }
 
+func updateMetricsTimes(componentIdForMetrics string, reconcileStartTime time.Time) {
```

Comment:
This function is about `set` not `update`.

Reply:
renamed